### PR TITLE
Jmap quotas draft v2

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: spec/quotas/quotas.mdown [hunk 1]
```diff
@@ -32,51 +35,95 @@ This document defines one additional capability URI.
...
+ignore any unknown data type in the list.
 
 The quota object MAY contain the following field:
 
-* **description**: `String|null` Arbitrary free, human readable, description of this quota. Might be used to explain where the limit comes from and explain the entities this quota applies to.
+* **warnLimit**: `UnsignedInt|null` The warn limit set by this quota object. It can be used to send a warning to an
+user that he is going to reach the hard limit soon, but no action is taken. If set, it should be lower than the
+`softLimit` and the `limit`.
+
+* **softLimit**: `UnsignedInt|null` The soft limit set by this quota object. It can be used to block outgoing messages,
```
Comment (chibenwa) [inline spec/quotas/quotas.mdown]: We need a non mail wording as it applies to the whole account. Some operations might be allowed, and some other defended?

File: spec/quotas/quotas.mdown [hunk 2]
```diff
@@ -32,51 +35,95 @@ This document defines one additional capability URI.
...
+* **limit**: `UnsignedInt` The hard limit set by this quota object. No more outgoing and ingoing messages should be
+allowed if we reach this limit. It should higher than the `warnLimit` and the `softLimit`.
+
+* **scope**: `Scope` The `Scope` of this quota.
 
-* **limit**: `UnsignedInt` The limit set by this quota object to the current mailbox.
+* **name**: `String` The name of the quota object. Useful for managing quotas and use queries for searching.
 
-* **limitScope**: `Scope` The `Scope` of the mailbox limit.
+* **datatypes**: `String[]` A list of all the data types values that are applying to this quota. This allows to assign
```
Comment (chibenwa) [inline spec/quotas/quotas.mdown]: Do we need a registry here? What if two extension defines the same type? Should we add the capability?

File: spec/quotas/quotas.mdown (hunk 1 above)
Comment (Arsnael) [reply spec/quotas/quotas.mdown]: You are right, it doesn't concern only messages anymore

Comment (judemille) [reply spec/quotas/quotas.mdown:108]: Is it normal in these sorts of specifications to refer to the user as `he`?

Comment (neilj) [reply spec/quotas/quotas.mdown:108]: No, the RFCs should be gender neutral. I'm sure the RFC editors would pick this up before publication but better to fix this now.

Comment (Arsnael) [reply spec/quotas/quotas.mdown:108]: I should really get back to that spec... Thanks for the feedback guys, it is indeed a mistake !

**Pull request merged**